core/modules/qdisp/JobQuery.h

```diff
@@ -0,0 +1,122 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ *
+ *  Created on: Sep 3, 2015
+ *      Author: jgates
+ */
+
+#ifndef LSST_QSERV_QDISP_JOBQUERY_H_
+#define LSST_QSERV_QDISP_JOBQUERY_H_
+
+// System headers
+#include <memory>
+#include <mutex>
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "qdisp/Executive.h"
+#include "qdisp/ResponseRequester.h"
+
+namespace lsst {
+namespace qserv {
+namespace qdisp {
+
+class QueryRequest;
+
+/** This class is used to describe, monitor, and control a single query to a worker.
+ *
+ */
+class JobQuery {
+public:
+    typedef std::shared_ptr<JobQuery> Ptr;
+    // Make a copy of the job description.
+    JobQuery(Executive* executive, JobDescription const& jobDescription,
+             JobStatus::Ptr jobStatus, std::shared_ptr<MarkCompleteFunc> markCompleteFunc) :
+        _executive(executive), _jobDescription(jobDescription),
+        _markCompleteFunc(markCompleteFunc), _jobStatus(jobStatus),
+        _attemptsToRun(0), _cancelled(false) {}
+
+    void setupWeakThis(Ptr jobQuery) {
+        _weakThis = jobQuery;
+        _jobDescription.respHandler()->setJobQuery(jobQuery);
+    }
+
+    virtual ~JobQuery() {
+        LOGF_DEBUG("~JobQuery _jobId=%1%" % getId());
+    }
+    int getAttemptsToRun() const {
+        std::lock_guard<std::recursive_mutex> lock(_rmutex);
+        return _attemptsToRun;
+    }
+    int getMaxRetries() const { return 5; }
```

Comment:
hmm, magic hidden number. Define as enum as private class variable instead, and write a comment why 5 is a reasonable number and what should one worry about if anything when changing that to say 2 or 100